### PR TITLE
changing open_cfradial1_datatree engine from 'netcdf4' to 'h5netcdf'

### LINKED ISSUE #194: ValueError when opening cfradial file using fsspec
* xradar version: 0.6.1
* Python version: 3.12.4
* Operating System: Ubuntu 22.04

Hi all, 

I am trying to open cfradial files from an `s3` bucket as follows,
```python 
import xradar as xd
import fsspec


def main():
    # Set the URL and path for the cloud
    URL = 'https://js2.jetstream-cloud.org:8001/'
    path = f'pythia/radar/erad2024'

    fs = fsspec.filesystem("s3", anon=True, client_kwargs=dict(endpoint_url=URL))
    files = fs.glob("pythia/radar/erad2024/20240522_MeteoSwiss_ARPA_Lombardia/Data/Cband/*.nc")
    radar_files = [f"s3://{i}" for i in files]
    with fs.open(radar_files[0], mode="rb") as fileObj:
        dt2 = xd.io.open_cfradial1_datatree(fileObj)


if __name__ == '__main__':
    main()

```

And I got the following error. 

```python 
Traceback (most recent call last):
  File "/media/alfonso/drive/Alfonso/python/raw2zarr/sigmet2zarr/delete.py", line 17, in <module>
    main()
  File "/media/alfonso/drive/Alfonso/python/raw2zarr/sigmet2zarr/delete.py", line 13, in main
    dt2 = xd.io.open_cfradial1_datatree(fs.open(radar_files[0]))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alfonso/mambaforge/envs/raw2zarr/lib/python3.12/site-packages/xradar/io/backends/cfradial1.py", line 347, in open_cfradial1_datatree
    ds = open_dataset(filename_or_obj, engine="netcdf4", **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alfonso/mambaforge/envs/raw2zarr/lib/python3.12/site-packages/xarray/backends/api.py", line 571, in open_dataset
    backend_ds = backend.open_dataset(
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/home/alfonso/mambaforge/envs/raw2zarr/lib/python3.12/site-packages/xarray/backends/netCDF4_.py", line 645, in open_dataset
    store = NetCDF4DataStore.open(
            ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alfonso/mambaforge/envs/raw2zarr/lib/python3.12/site-packages/xarray/backends/netCDF4_.py", line 381, in open
    raise ValueError(
ValueError: can only read bytes or file-like objects with engine='scipy' or 'h5netcdf'
```
Digging around, I found this is happening because the `netcdf4` engine does not support fsspec `s3.FileSystem` object  here

https://github.com/openradar/xradar/blob/482952b17f9a981fa94d8c989d200289ead96d04/xradar/io/backends/cfradial1.py#L345

However, if we change the engine to `h5netcdf`, it will work and will create exactly the same `datatree` object.

```python 
ds = open_dataset(filename_or_obj, engine="h5netcdf", **kwargs)

```
@kmuehlbauer, what do you think are the main implications of using `h5netcdf` instead of `netcdf4` engine?

### PR DESCRIPTION
**Pull request by aladinor**
- [ x] Closes #194 
- [ ] Changes are documented in `history.md`

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@aladinor I agree, this should be fixed. Could you please add `engine` as kwarg and use `netcdf4` as default engine?

**Comment by aladinor** [comment]
Sure!

File: xradar/io/backends/cfradial1.py [hunk 1]
```diff
@@ -327,22 +327,26 @@ def open_cfradial1_datatree(filename_or_obj, **kwargs):
         Import optional mandatory data and metadata, defaults to ``True``.
     site_coords : bool
         Attach radar site-coordinates to Dataset, defaults to ``True``.
+    engine: str
```
Comment (kmuehlbauer) [inline xradar/io/backends/cfradial1.py:330]: I think this should work for now. No immediate need for other types, like `BackendEntrypoint`. :+1:

**Review by kmuehlbauer — APPROVED** [review]
@aladinor Thanks, I think this is good to go. Please add a new `Development version` heading to history.md and list the your issue and your PR over there.

**Review by mgrover1 — APPROVED** [review]

**Review by kmuehlbauer — APPROVED** [review]

**Comment by kmuehlbauer** [comment]
Thanks @aladinor for the quick fix.

@mgrover1 Would be great to cut a release, if you find the time. :+1:

**Comment by mgrover1** [comment]
I can cut a release 👍

**Pull request merged**